### PR TITLE
docs: rework the homepage

### LINKED ISSUE #8856: docs: improvements to the home page
The Ibis project home page is better than it once was [citation needed], but
can use some improvements. In particular, it'd be great if we could have an
[interactive demo similar to
DuckDB's](https://shell.duckdb.org/#queries=v0,%20%20-Create-table-from-Parquet-file%0ACREATE-TABLE-train_services-AS%0A----FROM-'s3%3A%2F%2Fduckdb%20blobs%2Ftrain_services.parquet'~,%20%20-Get-the-top%203-busiest-train-stations%0ASELECT-station_name%2C-count(*)-AS-num_services%0AFROM-train_services%0AGROUP-BY-ALL%0AORDER-BY-num_services-DESC%0ALIMIT-3~).
This would required [PyArrow in
Pyodide](https://github.com/pyodide/pyodide/issues/2933) as the last blocker, I
think.

Regardless, we should ensure the landing page answers to a new/prospective user:

- What is Ibis?
- Why should I use Ibis?
- Confidence Ibis is a well-supported, production-ready library

Unfortunately, this may involve more HTML/CSS than I'm conformtable doing but
we'll figure something out.

### PR DESCRIPTION
**Pull request by lostmygithubaccount**
## Description of changes

This is a simple rework of the homepage. CSS/fancy things are ignored for now, but could be added in later. For now, this:

- adds a codespace button in the getting started tutorial
- small adjustment to sankey diagram (font size)
- makes the Ibis homepage have some of the stuff from the GitHub README
- explain a bit more and show some code examples
- unhide the quickstart from buttons
- other edits

## Issues closed

#8856

### CODE REVIEW

**Comment by lostmygithubaccount** [comment]
still should answer this, maybe:

> Confidence Ibis is a well-supported, production-ready library

**Comment by lostmygithubaccount** [comment]
some passing feedback from JIm (noting for later):

- rework titles/perhaps order to ensure portability comes across (i.e. not just fast local dataframes)
- remove local/remote example, perhaps just prose for that, too complicated
- remove quickstart
  - move some parts to getting started tutorial or other places
- CTAs at the bottom instead (getting started tutorial; why ibis; etc)

File: docs/index.qmd
Comment (deepyaman) [inline docs/index.qmd:141]: I don't particularly like the use of `backend` variable + f-string here; why have all the tabs and then use substitution, think you can pick one.

**Comment by lostmygithubaccount** [comment]
idk why the buttons at the bottom aren't full width, as they are locally. maybe scss stuff colliding? maybe if I add elsewhere

**Comment by lostmygithubaccount** [comment]
do NOT merge -- this does bad/unexpected things, will fix tomorrow

**Comment by lostmygithubaccount** [comment]
before merging, should also check w/ the 3 people quotes that they're okay with it. made some light edits to the quotes

File: docs/index.qmd
Comment (cpcloud) [inline docs/index.qmd:230]:
```suggestion
[Mark Druffel]{.blockquote-footer}
```

**Comment by lostmygithubaccount** [comment]
> Small edit: my last name is spelled Druffel, not druffle ☺️

sorry about that, thanks for catching!

**Review by cpcloud — APPROVED** [review]

**Comment by lostmygithubaccount** [comment]
thanks everyone! didn't get explicit approval from Daniel but given that quote is already one the website, will ask for forgiveness :)

**Comment by pybokeh** [comment]
Sorry, I missed your message Cody! @lostmygithubaccount   The quote looks good to me.

**Pull request merged**
